src/verification/mod.rs

```diff
@@ -118,6 +118,7 @@ pub(crate) struct JwtClaimsVerifier<'a, K>
 where
     K: JsonWebKey,
 {
+    allowed_jose_types: Option<Vec<JsonWebTokenType>>,
```

Comment:
since these should be unique, let's use a `HashSet` (following the example of `allowed_algs`)